### ISSUE TITLE
zabbix.server-mysql and zabbix.server-pgsql broken on master

### ISSUE THREAD

**Issue by aanderse** [NixOS/nixpkgs#95532]
**Describe the bug**

`zabbix.server-mysql` and `zabbix.server-pgsql` broke recently: https://hydra.nixos.org/build/122059780. I briefly poked around but I'm not sure what the problem or solution is. If anyone has any insight much appreciated.

@jonringer took a quick look as well and had [this](https://github.com/NixOS/nixpkgs/issues/95475#issuecomment-674441702) comment.

**To Reproduce**
1. `nix-build -A zabbix.server-mysql` on `master`, see build failure: `configure: error: LIBXML2 library not found`

**Expected behavior**

I expected the program to compile.

**Screenshots**

**Additional context**

**Notify maintainers**

@PsyanticY

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: zabbix.server-mysql zabbix.server-pgsql
# a list of nixos modules affected by the problem
module: services.zabbixServer
```

**Comment by jonringer** [comment]
removing the `--with-libxml2` configure flag allows for it to build, but I think that would be a regression in terms of server capabilities.

More proper fix would probably be to correct upstreams logic in how it finds libxml headers during configure

**Comment by aanderse** [comment]
@jonringer I'm under the impression this breakage wasn't introduced with a change in the `zabbix.server-mysql` package, so I'm left to believe something changed in `nixpkgs` which caused this. From your quick look is that not the case? :confused:

**Comment by jonringer** [comment]
I didn't look at any of the other packages, just investigated with `zabbix.server` which I guess points to `zabbix.server-pgsql`

**Comment by aanderse** [comment]
@jonringer sorry I should have clarified. My statement holds for both `zabbix.server-mysql` and `zabbix.server-pgsql`.

**Comment by aanderse** [comment]
I have double checked - both the `zabbix.server-pgsql` and `zabbix.server-mysql` packages from `master` compile on the `release-20.03` branch, so this isn't a change from `upstream` that directly caused it (maybe indirectly)... something happened in `nixpkgs` to cause this regression.

ping @mmahut and @PsyanticY as maintainers to take a look.

**Comment by pbogdan** [comment]
I found a PR that seems to be fixing a similar issue (https://github.com/NixOS/nixpkgs/pull/90249) after recent-ish pkg-config changes (https://github.com/NixOS/nixpkgs/pull/87705). I adapted the diff like so

```diff
diff --git a/pkgs/servers/monitoring/zabbix/server.nix b/pkgs/servers/monitoring/zabbix/server.nix
index fdf3fd476df..af7fc0cb04d 100644
--- a/pkgs/servers/monitoring/zabbix/server.nix
+++ b/pkgs/servers/monitoring/zabbix/server.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pkgconfig, curl, libevent, libiconv, libxml2, openssl, pcre, zlib
+{ stdenv, fetchurl, autoreconfHook, pkgconfig, curl, libevent, libiconv, libxml2, openssl, pcre, zlib
 , jabberSupport ? true, iksemel
 , ldapSupport ? true, openldap
 , odbcSupport ? true, unixODBC
@@ -25,7 +25,7 @@ in
         inherit sha256;
       };

-      nativeBuildInputs = [ pkgconfig ];
+      nativeBuildInputs = [ autoreconfHook pkgconfig ];
       buildInputs = [
         curl
         libevent
@@ -65,6 +65,14 @@ in
         find database -name data.sql -exec sed -i 's|/usr/bin/||g' {} +
       '';

+      preAutoreconf = ''
+        for i in $(find . -type f -name "*.m4"); do
+          substituteInPlace $i \
+            --replace 'test -x "$PKG_CONFIG"' 'type -P "$PKG_CONFIG" >/dev/null'
+        done
+      '';
+
+
       postInstall = ''
         mkdir -p $out/share/zabbix/database/
         cp -r include $out/
``
and that appears to fix the build but I can't confidently say it is the right solution.

**Comment by aanderse** [comment]
Thanks for your time and effort @pbogdan! :tada: 

@Ericson2314 how does this look to you?
@jonringer any thoughts?

**Comment by jonringer** [comment]
LGTM at a glance

**Comment by aanderse** [comment]
Resolved by #87712.
